### DISCUSSION TITLE
SIPS and STRP support

### DISCUSSION THREAD

**Discussion by svaza** [sipsorcery-org/sipsorcery#476] [Q&A]
Hi,

Does SIP Sorcery support SIPS and SRTP ?

Thanks

**Comment by sipsorcery** [comment]
SIPS yes. SRTP no.

SRTP is used in WebRTC connections but nobody has ever gotten around to it for SIP/VoIP. All the primitives are there if you feel like having a crack.

**Reply by svaza** [reply]
Thanks for replying, What are the other protocols supported by SipSorcery to ensure secured packet transfer between SIP trunk and client ?

**Reply by sipsorcery** [reply]
There are no media encryption options for SIP in this library.

For WebRTC connections media is encrypted using SRTP where the key gets negotiated over DTLS.

**Comment by svaza** [comment]
Can you give a headstart for supporting SRTP in SipSorcery ?

**Reply by sipsorcery** [reply]
If you look in the [DtlsSrtp](https://github.com/sipsorcery-org/sipsorcery/tree/master/src/net/DtlsSrtp) you'll see the classes related to SRTP. The high level steps would be something like:

 - Extract the SRTP logic from the classes mentioned to separate them from the DTLS handshake,
 - Determine which SIP key exchange mechanism you want to support,
 - Initialise the SRTP key from the new key exchange mechanism.
